lib/devise-security/models/session_limitable.rb

```diff
@@ -31,12 +31,71 @@ def update_unique_session_id!(unique_session_id)
         end
       end
 
+      # Removes old/inactive session if allowed.
```

Comment:
The purpose of `allow_limitable_authentication?` is to:
1. If `session_traceable` module is not included then it will always be `true`.
2. If didn't matched to 1 and number of sessions is still within the allowable number of session then it will allow authentication.
3. If didn't matched to 2 and will only reject authentication on limit then it will try to remove any expired session to accomodate new authentication using `deactivate_timeout_sessions!`.
4. If didn't matched to 3 (which reject authentication is disabled) then it will expire the oldest session regardless if its already expired.
5. If didn't matched to 4 then disallow authentication.

Reply:
Steps 3-5 don't feel like they belong in this method - boolean methods should only be returning true / false based on whatever logic you're adding. Steps 1-2 seem fine and what this method should be used for.